homeassistant/components/stream/hls.py

```diff
@@ -113,47 +156,137 @@ def render(track: StreamOutput) -> str:
             # don't autoplay. Also, hls.js uses the player parameter liveSyncDuration
             # which seems to take precedence for setting target delay. Yet it also
             # doesn't seem to hurt, so we can stick with it for now.
-            f"#EXT-X-START:TIME-OFFSET=-{EXT_X_START * track.target_duration:.3f}",
-        ]
+            playlist.append(
+                f"#EXT-X-START:TIME-OFFSET=-{EXT_X_START_NON_LL_HLS*track.target_duration:.3f},PRECISE=YES"
```

Comment:
Should this also be target_duration for consistency with the above reference?

Reply:
From the RFC, "The EXTINF duration of each Media Segment in a Playlist file, when rounded to the nearest integer, MUST be less than or equal to the Target Duration." and "The duration of a Partial Segment MUST be less than or equal to the Part Target Duration.".
The problem is that we don't have precise control over either of these: for the part durations, we are relying on ffmpeg to create partial segments using the `frag_duration` parameter we pass in, but there seems to be some minor variation in the resulting part durations. For segment durations, we have better control over when to cut a segment, but we still have to wait for the next keyframe. For the better surveillance cam codecs H264+/H265+, the keyframes can be quite far apart, so our segment durations may vary wildly. As a result, for both of these we don't know if the actual durations will be different than our targets, so to be safe we have to calculate the max of the segment/part durations each time. (We are doing this explicitly here for `part_duration`, but we are doing it implicitly when accessing the `track.target_duration` parameter).
Here's another wrinkle: the RFC spec says that nothing in the playlist should change except adding or removing segments and parts. If we follow this strictly we should just stick with fixed values here.
The good news is that things seem to mostly work as is, and they might work fine if we just used static values instead of worrying about this `max` business.

Reply:
To summarize the above, it looks like we will probably violate the RFC somehow. Either:
- The playlist segment duration and fragment duration might have a mismatch with the produced media, or
- We change the playlist segment duration and fragment duration with the media, but any such changes seem to be out of spec.

Neither of these seems to be too problematic. I pushed a commit that changes the behavior from the second option to the first option which is simpler.